### DISCUSSION TITLE
Create Login function in budibase

### DISCUSSION THREAD

**Discussion by subhro28** [Budibase/budibase#1975] [Help] [answered]
Hi everyone,

I was working on an app and i am stuck at creating **login** functionalities.

Can anyone please **guide** me on how to **create** **login functionality** in budibase ?

**Comment by mike12345567** [comment, marked as answer]
Hi @subhro28 Budibase is designed for building internal tools and as part of that we thought that it would be a waste of company resources to have to re-build the login functionality for every company app.

For this reason with Budibase every user logs into the Budibase platform itself, the same way that you logged in when you first started using it. When creating users through the Budibase portal (main screen before entering into an app) you'll be able to select if they have permission to build apps and if they do not then what apps they have permission to use.

You can also customise the login screen a bit with your organisation name and your company logo, you can input this information in the Organisation section of the Budibase portal.

**Reply by subhro28** [reply]
Hi @mike12345567 . Thank you for your response.

So, can i manage user's login credentials like emails along with passwords withing budibase predefined User database?

**Reply by mike12345567** [reply]
Yeap, if you look at the main page of your Budibase platform you'll see the users section of the portal, shown below:
![image](https://user-images.githubusercontent.com/4407001/125616780-817cd14c-d03d-4bb0-895f-9c64f9bd387c.png)

Here you can see a list of all users and then click into a user to give them access to various apps, roles within apps etc. You can also invite new users through our email onboarding system shown below:
![image](https://user-images.githubusercontent.com/4407001/125616911-50ad2587-2ab9-4ba3-bdc7-a5c7f7699b67.png)

We are planning to expand the user management capabilities here in the future with the addition of groups/workspaces, which will allow grouping users together to give them access to a set of apps/roles and so on. This will mean adding a new user is very easy, you simply invite them and add them to the workspaces that they need to have access to.

**Reply by subhro28** [reply]
Thank you @mike12345567 .

One last thing to ask. Can i create a form for users to sign up and login to my app using the same User database?

**Reply by mike12345567** [reply]
This is something that we have steered away from, as Budibase is designed for companies' internal tools we designed it such that a company administrator would manage all of the various sign up tasks, the idea being that when a new employee is being onboarded, or changing teams you could simply send an invitation email when setting up their email address.

**Reply by SantoshSrinivas79** [reply]
@mike12345567 could you pls advise how to automate user creation? Basically, I am thinking of a webhook that would create users automatically along with role access for that user for an app .... say after a HR onboarding application.

I believe this was possible earlier. Please advise.

**Reply by mjashanks** [reply]
@SantoshSrinivas79 our API is useable, but not supported or documented. Your best bet is to watch network requests when manually creating users, and mimic them.

We removed the "Create User" automation step, thus the ability to build a Self-signup, for the reasons above.

**Reply by marceldarvas** [reply]
With the API I was looking at the create user request, the parameters don't specify the user welcome/invite mail.
If we are planning to use OpenID provider, is there way to skip that alert? Or should I look at SMTP relay filtering?

**Reply by shogunpurple** [reply]
@marceldarvas creating users through the API bypasses the whole welcome/invite email process. Its mainly used for backfilling or creating users manually through scripts, rather than a way to trigger the same onboarding process you get from invitation in the UI.
